content/en/docs/concepts/configuration/secret.md

```diff
@@ -819,74 +529,16 @@ field set to that of the service account.
-of very large secrets which would exhaust the API server and kubelet memory.
-However, creation of many smaller secrets could also exhaust memory. More
-comprehensive limits on memory usage due to secrets is a planned feature.
-
-The kubelet only supports the use of secrets for Pods where the secrets
-are obtained from the API server.
-This includes any Pods created using `kubectl`, or indirectly via a replication
-controller. It does not include Pods created as a result of the kubelet
-`--manifest-url` flag, its `--config` flag, or its REST API (these are
-not common ways to create Pods). 
-The `spec` of a {{< glossary_tooltip text="static Pod" term_id="static-pod" >}} cannot refer to a Secret
-or any other API objects.
-
-
-Secrets must be created before they are consumed in Pods as environment
-variables unless they are marked as optional. References to secrets that do
-not exist will prevent the Pod from starting.
+### Using Secrets with static Pods {#restriction-static-pod}
 
-References (`secretKeyRef` field) to keys that do not exist in a named Secret
-will prevent the Pod from starting.
-
-Secrets used to populate environment variables by the `envFrom` field that have keys
-that are considered invalid environment variable names will have those keys
-skipped. The Pod will be allowed to start. There will be an event whose
-reason is `InvalidVariableNames` and the message will contain the list of
-invalid keys that were skipped. The example shows a pod which refers to the
-default/mysecret that contains 2 invalid keys: `1badkey` and `2alsobad`.
-
-```shell
-kubectl get events
-```
-
-The output is similar to:
-
-```
-LASTSEEN   FIRSTSEEN   COUNT     NAME            KIND      SUBOBJECT                         TYPE      REASON
-0s         0s          1         dapi-test-pod   Pod                                         Warning   InvalidEnvironmentVariableNames   kubelet, 127.0.0.1      Keys [1badkey, 2alsobad] from the EnvFrom secret default/mysecret were skipped since they are considered invalid environment variable names.
-```
-
-### Secret and Pod lifetime interaction
-
-When a Pod is created by calling the Kubernetes API, there is no check if a referenced
-secret exists. Once a Pod is scheduled, the kubelet will try to fetch the
-secret value. If the secret cannot be fetched because it does not exist or
-because of a temporary lack of connection to the API server, the kubelet will
-periodically retry. It will report an event about the Pod explaining the
-reason it is not started yet. Once the secret is fetched, the kubelet will
-create and mount a volume containing it. None of the Pod's containers will
-start until all the Pod's volumes are mounted.
+You cannot use ConfigMaps or Secrets with
+{{< glossary_tooltip text="static Pods" term_id="static-pod" >}}.
 
 ## Use cases
 
-### Use-Case: As container environment variables
+### Use case: As container environment variables
```

Comment:
Consider adding text between the level 2 heading that the first level 3 heading to introduce the use cases below.

Reply:
I'm very happy to save that for another PR - this one is making substantial enough changes that I don't see the need to cover every shortcoming we can spot.

Reply:
Not a problem. You have already done a ton of work here.